tests/unit/contentproviders/test_git.py

```diff
@@ -18,6 +19,28 @@ def test_clone(repo_with_content):
         assert git_content.content_id == sha1[:7]
 
 
+def test_submodule_clone(repo_with_submodule):
+    """Test git clone containing a git submodule."""
+    upstream, sha1_upstream, sha1_submod = repo_with_submodule
```

Comment:
Maybe we can prefix these with `expected_`? I think the similar variable names to the ones we get from the actually checked out repo are the source of the `assert` statement mixup below. Definitely took me a while to spot that `sha1_submod` and `submod_sha1` where not the same name :)